ext/nio4r/org/nio4r/ByteBuffer.java

```diff
@@ -0,0 +1,151 @@
+package org.nio4r;
+
+import org.jruby.Ruby;
+import org.jruby.RubyClass;
+import org.jruby.RubyNumeric;
+import org.jruby.RubyObject;
+import org.jruby.anno.JRubyMethod;
+import org.jruby.javasupport.JavaUtil;
+import org.jruby.runtime.ThreadContext;
+import org.jruby.runtime.builtin.IRubyObject;
+
+/*
+
+created by Upekshej
+
+ */
+
+public class ByteBuffer extends RubyObject {
+
+    private java.nio.ByteBuffer byteBuffer;
+
+	public ByteBuffer(final Ruby ruby, RubyClass rubyClass) {
+            super(ruby, rubyClass);
+    }
+
+    @JRubyMethod (name= "allocate=")
+    public IRubyObject allocate(ThreadContext context,IRubyObject size){
+        int allocationSize = RubyNumeric.num2int(size);
+        try {
+            byteBuffer = java.nio.ByteBuffer.allocate(allocationSize);
+        }catch(IllegalArgumentException ie){
+            throw context.runtime.newArgumentError("IllegalArgumentException: " + ie.getLocalizedMessage());
+        }catch (Exception e){
+            throw context.runtime.newRuntimeError(e.getLocalizedMessage());
+        }
+        return context.nil;
+    }
+
+    @JRubyMethod (name = "wrap=")
+    public IRubyObject wrap(ThreadContext context, IRubyObject string,IRubyObject offset, IRubyObject limit){
+        String stringToWrap = string.asJavaString();
+        byte[] byteArray =stringToWrap.getBytes();
+        int arrayOffset = RubyNumeric.num2int(offset);
+        int arrayLimit = RubyNumeric.num2int(limit);
+        byteBuffer = java.nio.ByteBuffer.wrap(byteArray,arrayOffset,arrayLimit);
+        return context.nil;
+    }
+
+    @JRubyMethod (name = "wrap=")
+    public IRubyObject wrap(ThreadContext context, IRubyObject string){
+        String stringToWrap = string.asJavaString();
+        byte[] byteArray =stringToWrap.getBytes();
+        byteBuffer = java.nio.ByteBuffer.wrap(byteArray);
+        return context.nil;
+    }
+
+    /**
+     * Currently assuming only strings will come..
+     *
+     * @param context
+     * @return
+     */
+    @JRubyMethod
+    public IRubyObject putAByte(ThreadContext context, IRubyObject aByte){
+        byte b = (byte)aByte.toJava(byte.class);
+        byteBuffer.put(b);
+        return context.nil;
+    }
+
+    //see http://www.programcreek.com/java-api-examples/index.php?api=org.jruby.runtime.builtin.IRubyObject.
+    @JRubyMethod
+    public IRubyObject putByteArray(ThreadContext context, IRubyObject aByte){
```

Comment:
We should pick more idiomatic Ruby names for these. I think `#put` with a `<<` alias is the way to go.

`aByte` should probably just be `str`, and this API should be used for adding strings to ByteBuffers.